app/api/routes/product_routes.py

```diff
@@ -0,0 +1,70 @@
+from datetime import datetime
+from typing import List
+from uuid import UUID
+from fastapi import APIRouter, HTTPException, Depends
+
+from app.models import User 
+from app.product_model import Product, CreateProductRequest, UpdateProductRequest
+from app.api.auth import get_current_admin
+
+router = APIRouter()
+
+products_db = {
+    UUID("1e2d4567-e89b-12d3-a456-426614174000"): {
+        "id": UUID("1e2d4567-e89b-12d3-a456-426614174000"),
+        "name": "Product A",
+        "description": "Description of Product A",
+        "price": 100.0,
+        "created_at": datetime.now(),
+        "updated_at": datetime.now(),
+    },
+    UUID("2f3d4567-e89b-12d3-a456-426614174001"): {
+        "id": UUID("2f3d4567-e89b-12d3-a456-426614174001"),
+        "name": "Product B",
+        "description": "Description of Product B",
+        "price": 150.0,
+        "created_at": datetime.now(),
+        "updated_at": datetime.now(),
+    },
+}
+
+# Create product (only admin)
+@router.post("/products", response_model=Product)
+async def create_product(product: CreateProductRequest, current_admin: User = Depends(get_current_admin)):
+    new_product = Product(**product.dict(), created_at=datetime.now(), updated_at=datetime.now())
```

Comment:
Update date for a new product?

Reply:
I set the updated_at date to reflect the last time a modification was made. During the creation of a new product, the updated_at date is set to the same value as created_at, which means it takes the current time as the creation date

Reply:
Yes, that's why it's invalid. The update date is the last update date, and it should only be changed in the update endpoint.